django-s3-file-field-client/src/client.ts

```diff
@@ -25,10 +25,19 @@ export interface UploadResult {
   state: 'aborted' | 'successful' | 'error';
 }
 
-export default class S3FFClient {
-  protected readonly baseUrl: string;
+export interface ProgressEvent {
+  loaded?: number;
+  total?: number;
+  state: 'initializing' | 'sending' | 'finalizing';
```

Comment:
Could this value be an `enum`?

Reply:
I was following the pattern from line 25 above. For the sake of consistency, we should use the same pattern in both places, whichever we choose. (I leave it up to you to pick which one is best.)

Reply:
I see. For TypeScript, I think `enum` is the right choice in both places.